### ISSUE TITLE
deprecated code elements so soon in the book's lifecycle?

### ISSUE THREAD

**Issue by matanster** [concurrent-programming-in-scala/learning-examples#7]
```
[info] Set current project to concurrency-examples
> set scalacOptions in ThisBuild ++= Seq("-deprecation")
[info] Defining {.}/*:scalacOptions
[info] The new value will be used by compile:scalacOptions
[info] Reapplying settings...
[info] Set current project to concurrency-examples (in build file:/repos/learning-examples/)
> compile
[info] Compiling 50 Scala sources to /repos/learning-examples/target/scala-2.11/classes...
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch4/Promises.scala:131: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.
[warn]         signature: Promise.success(value: T): Promise.this.type
[warn]   given arguments: <none>
[warn]  after adaptation: Promise.success((): Unit)
[warn]       def run() = p success ()
[warn]                     ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch4/Promises.scala:176: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.
[warn]         signature: Promise.trySuccess(value: T): Boolean
[warn]   given arguments: <none>
[warn]  after adaptation: Promise.trySuccess((): Unit)
[warn]   cancel trySuccess ()
[warn]          ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch4/Futures.scala:183: match may not be exhaustive.
[warn] It would fail on the following input: Failure(_)
[warn]   longestBuildLine onComplete {
[warn]                               ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch3/Collections.scala:27: trait BufferProxy in package mutable is deprecated: Proxying is deprecated due to lack of use and compiler-level support.
[warn]   val buffer = new mutable.BufferProxy[Int] with mutable.SynchronizedBuffer[Int] {
[warn]                            ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch3/Collections.scala:27: trait SynchronizedBuffer in package mutable is deprecated: Synchronization via traits is deprecated as it is inherently unreliable.  Consider java.util.concurrent.ConcurrentLinkedQueue as an alternative.
[warn]   val buffer = new mutable.BufferProxy[Int] with mutable.SynchronizedBuffer[Int] {
[warn]                                                          ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch3/Processes.scala:47: method lines_! in trait ProcessBuilder is deprecated: Use lineStream_! instead.
[warn]     val proclines = "ps -A".lines_!
[warn]                             ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch3/Processes.scala:67: method lines_! in trait ProcessBuilder is deprecated: Use lineStream_! instead.
[warn]   def files(pattern: String): Stream[String] = s"find /home/ -name $pattern".lines_!
[warn]                                                                              ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch3/Processes.scala:79: method lines_! in trait ProcessBuilder is deprecated: Use lineStream_! instead.
[warn]     val proclines = "ps -A" #| s"grep $pattern" lines_!;
[warn]                                                 ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch6/Composition.scala:78: method create in object Observable is deprecated: Use `apply[T](Subscriber[T] => Unit)` instead
[warn]   def randomQuote = Observable.create[String] { obs =>
[warn]                                ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch6/Observables.scala:57: method create in object Observable is deprecated: Use `apply[T](Subscriber[T] => Unit)` instead
[warn]   val vms = Observable.create[String] { obs =>
[warn]                        ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch6/Observables.scala:82: method create in object Observable is deprecated: Use `apply[T](Subscriber[T] => Unit)` instead
[warn]   val o = Observable.create[String] { obs =>
[warn]                      ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch6/Observables.scala:130: method create in object Observable is deprecated: Use `apply[T](Subscriber[T] => Unit)` instead
[warn]     Observable.create { observer =>
[warn]                ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch6/Observables.scala:168: method create in object Observable is deprecated: Use `apply[T](Subscriber[T] => Unit)` instead
[warn]     Observable.create { observer =>
[warn]                ^
[warn] /repos/learning-examples/src/main/scala/org/learningconcurrency/ch9/package.scala:54: method create in object Observable is deprecated: Use `apply[T](Subscriber[T] => Unit)` instead
[warn]     Observable.create { obs =>
[warn]                ^
[warn] there were 4 feature warning(s); re-run with -feature for details
[warn] 15 warnings found

> about
[info] This is sbt 0.13.7
[info] The current project is ... learning-examples 1.0
[info] The current project is built against Scala 2.11.1
[info] Available Plugins: sbt.plugins.IvyPlugin, sbt.plugins.JvmPlugin, sbt.plugins.CorePlugin, sbt.plugins.JUnitXmlReportPlugin, com.typesafe.sbteclipse.plugin.EclipsePlugin
[info] sbt, sbt plugins, and build definitions are using Scala 2.10.4
```

Or should it rather be compiled differently or with certain Scala versions for the sbt build definitions?

**Comment by axel22** [comment]
Good point - we should fix this in the code. Then we can get back to Packt, so that they can create an 'errata' document.

**Comment by axel22** [comment]
The automatic adaptation for `()` arguments is getting deprecated in Scala - this is now fixed.

`SynchronizedBuffer`s are deprecated in Scala now, but this example is only in this repo, and not in the book. I've commented it out.

Rx is in a constant state of change, and much of its API is not yet stable.
However, the differences are mostly in the method names - the code in this repo has now been updated.
